lib/chrono_logger.rb

```diff
@@ -75,7 +75,7 @@ def initialize(log = nil, opt = {})
     end
 
     def write(message)
-      check_and_shift_log if @pattern
+      check_and_shift_log if defined? @pattern
```

Comment:
Maybe it cause performance issue because this method is called at all the log output. How about setting it with `nil` on `initialize`.

Reply:
Do you mean following?
```ruby
check_and_shift_log unless @pattern.nil?
```
I think this pattern call `nil?` method too.
I think it is no difference to call `defined?`.

Reply:
Sorry for the lack of information. I meant to initialize `@pattern` with `nil` at `TimeBasedLogDevice#initialize`. For example like the following code. This way can avoid the performance issue on every call of `write`.

```
  class TimeBasedLogDevice < LogDevice
    ...
    def initialize(log = nil, opt = {}) 
      @dev = @filename = @pattern = nil
      ...
    end
```